### PR TITLE
Fix/issue#683

### PR DESCRIPTION
**Pull request by PattyDePuh**
## Issue [#683](https://github.com/VROOM-Project/vroom/issues/683)

Created an accessor for the `has_custom_location_index` variable inside `Input`.
Accessor is used at the end of `parse()`, if no custom matrizes where found inside JSON input.

### CODE REVIEW

**Review by jcoupey — CHANGES_REQUESTED** [review]
Thanks for submitting a PR!

The changes you made would work when parsing a json problem with `location_index` values and no matrix but it would still be possible to have inconsistent definitions when using the C++ API.
This is pretty much the same as in #628: we should enforce the checks at a lower level in the `Input` class so that they apply both to json-defined problems and uses of the C++ API.

A good place to do this is at the beginning of [`Input::set_matrices`](https://github.com/VROOM-Project/vroom/blob/39438be9e5b61d88a5608a71b369994fc212f49a/src/structures/vroom/input/input.cpp#L634) because this function is called after the problem definition, prior to solving. There is already a check for missing location index there.

**Comment by PattyDePuh** [comment]
Thanks for Feedback.

File: src/utils/input_parser.cpp [hunk 1]
```diff
@@ -580,13 +580,13 @@ void parse(Input& input, const std::string& input_str, bool geometry) {
...
+  } else if (json_input.HasMember("matrix")) {
     // Deprecated `matrix` key still interpreted as
     // `matrices.DEFAULT_PROFILE.duration` for retro-compatibility.
-    if (json_input.HasMember("matrix")) {
-      input.set_durations_matrix(DEFAULT_PROFILE,
-                                 get_matrix<Duration>(json_input["matrix"]));
-    }
+    input.set_durations_matrix(DEFAULT_PROFILE,
+                                get_matrix<Duration>(json_input["matrix"]));
+  } else {
```
Comment (jcoupey) [inline src/utils/input_parser.cpp:588]: Not really a fan of a logic branch that does nothing, and I don't think we really need to comment here on stuff that other parts of the code is responsible for. We can probably revert all changes to `input_parser.cpp`?

**Review by jcoupey — COMMENTED** [review]
Thanks for the changes, the check in `Input::set_matrices` does the job! Can you please add a changelog entry under the unreleased/fixed section?

**Comment by PattyDePuh** [comment]
Everything is cleaned up and ready to merge.

**Review by jcoupey — APPROVED** [review]

**Pull request merged**